### PR TITLE
Feat (Auth): Integration Tests for fetchCurrentDevice API

### PR DESCRIPTION
**Pull request by hahnandrew**
- **feat: added integration test suite**
- **feat: added methods for integration test suite**

*Description of changes:*

Added integration test suites for `fetchCurrentDevice` API

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/auth/amplify_auth_cognito/example/integration_test/device_tracking_test.dart [hunk 1]
```diff
@@ -240,6 +240,45 @@ void main() {
...
+        final previousDeviceKey = await getDeviceKey();
+        await signOutUser();
+        await deleteDevice(cognitoUsername, previousDeviceKey!);
+        await signIn();
+        final newCurrentTestDevice = await Amplify.Auth.fetchCurrentDevice();
+        expect(newCurrentTestDevice.id, isNot(previousDeviceKey));
+      });
+
+      asyncTest(
+          'fetchCurrentDevice throws a DeviceNotTrackedException when device is forgotton.',
```
Comment (khatruong2009) [inline packages/auth/amplify_auth_cognito/example/integration_test/device_tracking_test.dart:262]:
```suggestion
          'fetchCurrentDevice throws a DeviceNotTrackedException when device is forgotten.',
```

**Review by khatruong2009 — APPROVED** [review]

File: packages/auth/amplify_auth_cognito/example/integration_test/device_tracking_test.dart (hunk 1 above)
Comment (hahnandrew) [reply packages/auth/amplify_auth_cognito/example/integration_test/device_tracking_test.dart]: [fixed] (https://github.com/aws-amplify/amplify-flutter/pull/5072/commits/627c3f15186751ea158a00a58a69a70211a322b6)

Comment (hahnandrew) [reply packages/auth/amplify_auth_cognito/example/integration_test/device_tracking_test.dart]: [fixed](https://github.com/aws-amplify/amplify-flutter/pull/5072/commits/627c3f15186751ea158a00a58a69a70211a322b6)

**Review by Equartey — APPROVED** [review]
LGTM 🚢

**Pull request merged**